src/Optimization/hiopKKTLinSysSparse.cpp

```diff
@@ -295,15 +298,18 @@ namespace hiop
 #endif  // HIOP_USE_STRUMPACK        
         }
 
-        if(linSys_) {
+        if( (nullptr == linSys_ && linear_solver == "auto") || linear_solver == "ginkgo") {
+          //ma57, pardiso and strumpack are not available or user requested ginkgo
+#ifdef HIOP_USE_GINKGO              
           nlp_->log->printf(hovScalars,
-                            "KKT_SPARSE_XYcYd linsys: alloc [%s] matrix size %d (%d cons)(cpu)\n",
-                            linsol_actual.c_str(),
-                            n,
-                            neq+nineq);
+                            "KKT_SPARSE_XYcYd linsys: alloc GINKGO with matrix size %d (%d cons)\n",
+                            n, neq+nineq);
+          hiopLinSolverSymSparseGinkgo *p = new hiopLinSolverSymSparseGinkgo(n, nnz, nlp_);
+          p->setFakeInertia(neq + nineq);
+          linSys_ = p;        
```

Comment:
Fake inertia is deprecated, can be safely removed from the code
```suggestion
          linSys_ = new hiopLinSolverSymSparseGinkgo(n, nnz, nlp_);
```